Source/missiles.cpp

```diff
@@ -4269,27 +4269,30 @@ void MI_Rhino(int i)
 		return;
 	}
 	GetMissilePos(i);
-	Point position = missile[i].position.tile;
-	dMonster[position.x][position.y] = 0;
+	Point prevPos = missile[i].position.tile;
+	Point newPosSnake;
+	dMonster[prevPos.x][prevPos.y] = 0;
 	if (monster[monst]._mAi == AI_SNAKE) {
 		missile[i].position.traveled += missile[i].position.velocity * 2;
 		GetMissilePos(i);
+		newPosSnake = missile[i].position.tile;
 		missile[i].position.traveled -= missile[i].position.velocity;
 	} else {
 		missile[i].position.traveled += missile[i].position.velocity;
 	}
 	GetMissilePos(i);
-	if (!PosOkMonst(monst, position) || (monster[monst]._mAi == AI_SNAKE && !PosOkMonst(monst, position))) {
-		MissToMonst(i, position);
+	Point newPos = missile[i].position.tile;
+	if (!PosOkMonst(monst, newPos) || (monster[monst]._mAi == AI_SNAKE && !PosOkMonst(monst, newPosSnake))) {

```

Comment:
Good pickup, I should've realised the side effects in `GetMissilePos()` modify `missile[i].position.tile` and make the order of reads important -_- (plus the special case needed for AI_SNAKE checking two ticks/moves ahead)

Reply:
That's not so easy to detected.
The Method `GetMissilePos` is really misleading.
We should change it to something like `UpdateMissilePos` (not part of this pr).